fastapi/security/http.py

```diff
@@ -158,8 +158,11 @@ async def __call__(
             else:
                 return None
         if scheme.lower() != "digest":
-            raise HTTPException(
-                status_code=HTTP_403_FORBIDDEN,
-                detail="Invalid authentication credentials",
-            )
+            if self.auto_error:
+                raise HTTPException(
+                    status_code=HTTP_403_FORBIDDEN,
+                    detail="Invalid authentication credentials",
+                )
+            else:
```

Comment:
No need the statement `else` here.

Reply:
Good point, I removed the extra statements in the rest of the file as well.